revoke.sharedmailboxes.ps1

```diff
@@ -69,576 +55,331 @@ function Resolve-HTTPError {
+        $accessToken = $Response.access_token
+
+        # Connect to Exchange Online in an unattended scripting scenario using an access token.
+        Write-Verbose "Connecting to Exchange Online"
+
+        $exchangeSessionParams = @{
+            Organization     = $AADOrganization
+            AppID            = $AADAppID
+            AccessToken      = $accessToken
+            CommandName      = $commands
+            ShowBanner       = $false
+            ShowProgress     = $false
+            TrackPerformance = $false
+            ErrorAction      = "Stop"
+        }
+        $exchangeSession = Connect-ExchangeOnline @exchangeSessionParams
+        
+        Write-Information "Successfully connected to Exchange Online"
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
 
-                $dryRun = $using:dryRun
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        $auditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error connecting to Exchange Online. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $True
+            })
 
-                # Create array for logging since the "normal" Write-Information isn't sent to HelloID as another PS session performs the commands
-                $verboseLogs = [System.Collections.ArrayList]::new()
-                $informationLogs = [System.Collections.ArrayList]::new()
-                $warningLogs = [System.Collections.ArrayList]::new()
-                    
-                # Import module
-                $moduleName = "ExchangeOnlineManagement"
-                $commands = $using:commands
+        # Skip further actions, as this is a critical error
+        continue
+    }
 
-                # If module is imported say that and do nothing
-                if (Get-Module | Where-Object { $_.Name -eq $ModuleName }) {
-                    [Void]$verboseLogs.Add("Module $ModuleName is already imported.")
-                }
-                else {
-                    # If module is not imported, but available on disk then import
-                    if (Get-Module -ListAvailable | Where-Object { $_.Name -eq $ModuleName }) {
-                        $module = Import-Module $ModuleName -Cmdlet $commands
-                        [Void]$verboseLogs.Add("Imported module $ModuleName")
+    # revoke Exchange Online Mailbox permission
+    foreach ($permission in $pRef.Permissions) {
+        switch ($permission) {
+            "Full Access" {
+                try {
+                    Write-Verbose "Revoking permission [FullAccess] to mailbox [$($pRef.Name) ($($pRef.id))] for user [$($aRef.UserPrincipalName) ($($aRef.Guid))]"
+
+                    $FullAccessPermissionSplatParams = @{
```

Comment:
Update SplatParms to:

                    Identity        = $pRef.id
                    User            = $aRef.Guid
                    AccessRights    = 'FullAccess'
                    InheritanceType = 'All'
                    Confirm         = $false

Reply:
Updated